woocommerce/woocommerce-ios
# [Reliability] Add failsafe decoding when String is received for stock_quantity

Closes: #9840


## Description

This PR increases the app's resilience when the app receives a `String` instead of an expected value type for the `stockQuantity` on `Product` or `ProductVariation`. (This is a known decoding error that users have experienced.)

Core sends the stock quantity as an `Int` or `null`, and we already convert that to an optional `Decimal` to handle plugins that allow decimal quantities. This PR adds failsafe decoding from `String` to `Decimal`, too.

## Testing instructions

We don't know the exact plugins that can cause this behavior, but you can test using a tool like Charles Proxy or Proxyman to intercept and modify the response:

1. Set a breakpoint in your tool of choice for requests to the `/wc/v3/products` endpoint.
2. Build and run the app.
3. Open the Products tab.
4. Modify the response so a product's `stock_quantity` field returns a string representing a number or null value (e.g. `"stock_quantity": "1"` or `"stock_quantity": ""`).
5. Execute the response and confirm the app loads the product list.

You can repeat the same steps with the variations list for a variable product, modifying the same fields in the response for a variation.

---
